### PR TITLE
Improve PowerShell 7 support

### LINKED ISSUE #4911: M365DSCDeltaReport (Compare Configuration files) does not work
### Description of the issue

DeltaReport functionality does not work as intented. Exported Configuration files from Entra ID tenant on two different instances. Changes were made between both instances. Next, the New-M365DSCDeltaReport cmdlet was used to compare the two files and report it in an hmtl file. the Function did not detect any drift and displayed " No discrepancies were found" 

### Microsoft 365 DSC Version

DEV / 1.24.717.1

### Which workloads are affected

Azure Active Directory (Entra ID)

### The DSC configuration

```powershell
DeltaReport/Comparing Configurations
https://microsoft365dsc.com/user-guide/get-started/comparing-configurations/
```

### Verbose logs showing the problem

_No response_

### Environment Information + PowerShell Version

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.3
PSEdition                      Core
GitCommitId                    7.4.3
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by FabienTschanz**
#### Pull Request (PR) description
This Pull Request attempts to improve the support of PowerShell 7, especially the handling of the `PSDesiredStateConfiguration` module in version 1.0 with the cmdlet `Get-DscResource`. In PowerShell 7, the version 1.0 of this module does not return the DSC resources formatted properly. For this to work properly, the version 2.0.7 must be installed. 

This PR introduces a new module dependency to `PSDesiredStateConfiguration 2.0.7` and installs this one, if PowerShell Core is used. It then loads the cmdlets of this module with the Prefix `Pwsh` to make sure no naming conflicts happen with version 1.0. 
For this prefix to work, the Microsoft365DSC modules using cmdlets of the `PSDesiredStateConfiguration` module must check which PowerShell version they are running and then use the appropriate cmdlet. 

This PR depends on https://github.com/microsoft/DSCParser/pull/50 for the best experience.

#### This Pull Request (PR) fixes the following issues
- Fixes #4911

### CODE REVIEW

File: docs/docs/user-guide/get-started/powershell7-support.md [hunk 1]
```diff
@@ -41,10 +41,10 @@ that location or use PowerShell 5.1 to install the modules using 'Install-Module
...
 
-# PSDesiredStateConfiguration Needs to be Installed Separately
+# PSDesiredStateConfiguration needs to be installed separately
 
-Starting with PowerShell 7.2, the core Desired State Configuration module (PSdesiredStateConfiguration) has been decoupled from the core PowerShell build and now need to be installed separately. In a PowerShell 7+ console, you can install the module by running the command:
+Starting with PowerShell 7.2, the core Desired State Configuration module (PSdesiredStateConfiguration) has been decoupled from the core PowerShell build and now needs to be installed separately. In a PowerShell 7+ console, you can install the module by running the command:
 
 ```powershell
-Install-Module PSDesiredStateConfiguration -Force
+Install-Module PSDesiredStateConfiguration -RequiredVersion 2.0.7 -Force
```
Comment (NikCharlebois) [inline docs/docs/user-guide/get-started/powershell7-support.md]: I think it's either we add it to the manifest and remove this instruction from here, or leave it out of manifest and keep it here. I am afraid we'll end up having different versions listed in documentation and manifest otherwise.

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: docs/docs/user-guide/get-started/powershell7-support.md (hunk 1 above)
Comment (FabienTschanz) [reply docs/docs/user-guide/get-started/powershell7-support.md]: Okay, just a moment. Let me rephrase it so we don't have any ambiguity there.

Comment (FabienTschanz) [reply docs/docs/user-guide/get-started/powershell7-support.md]: Updated, now using `Update-M365DSCDependencies -Scope AllUsers`, which in itself goes to the manifest and installs the module if it isn't found in the current PowerShell 7 session.

Comment (FabienTschanz) [reply docs/docs/user-guide/get-started/powershell7-support.md]: @NikCharlebois Can you please have a look at https://github.com/microsoft/DSCParser/pull/50 as well?

**Review by NikCharlebois — APPROVED** [review]
